### PR TITLE
Internal Review][CL][deletePosition]: test edge cases where one of the indexes does not exist

### PR DESCRIPTION
**Pull request by stackman27**
Closes: [#5052](https://github.com/osmosis-labs/osmosis/issues/5052)

## What is the purpose of the change

In deletePosition, we delete from multiple indexes. However, there is only one error case covered by tests - PositionIdNotFoundError

## Testing and Verifying
added test

### CODE REVIEW

File: x/concentrated-liquidity/position.go
Comment (stackman27) [inline x/concentrated-liquidity/position.go:306]: note: wondering if the order of deletion matters?  because i had to move this down because, we would never hit `PoolPositionIdNotFoundError` case because the `AddressPoolPositionIdNotFoundError` would always error first. 

i can move it back up but it'll be a slightly more complex test.

Comment (p0mvn) [reply x/concentrated-liquidity/position.go:306]: It doesn't. If error occurs, any state updates are reverted in `runTx` in base app

File: x/concentrated-liquidity/position_test.go
Comment (p0mvn) [inline x/concentrated-liquidity/position_test.go:669]: Consider setting `tc.poolId` to the desired value in the test case instead of modifying here

File: x/concentrated-liquidity/position_test.go
Comment (p0mvn) [inline x/concentrated-liquidity/position_test.go:588]:
```suggestion
			name:             "Invalid case: non-existent address-pool-position ID to position mapping",
```

File: x/concentrated-liquidity/position_test.go
Comment (p0mvn) [inline x/concentrated-liquidity/position_test.go:595]:
```suggestion
			name:             "Invalid case: non-existent pool-position ID mapping",
```

**Review by p0mvn — APPROVED** [review]
LGTM after comments are addressed

File: x/concentrated-liquidity/position_test.go
Comment (p0mvn) [inline x/concentrated-liquidity/position_test.go:675]: What is the purpose of replacing defautls? We should be able to simply do:
```go
err = s.App.ConcentratedLiquidityKeeper.DeletePosition(s.Ctx, test.positionId, defaultCreator, defaultPoolId)
```

and se the desired `test.poolId` and `test.creator` for each test case

**Review by p0mvn — CHANGES_REQUESTED** [review]
Would like to converge on my latest comment before merging

File: x/concentrated-liquidity/position_test.go
Comment (stackman27) [reply x/concentrated-liquidity/position_test.go]: yea that works too. changed

**Review by p0mvn — APPROVED** [review]
LGTM

**Pull request merged**
